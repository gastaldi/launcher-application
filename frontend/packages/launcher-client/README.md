# launcher-client [![Build Status](https://travis-ci.com/fabric8-launcher/launcher-client.svg?branch=master)](https://travis-ci.com/fabric8-launcher/launcher-client)
Typescript / Javascript client to communicate with the launcher backends
