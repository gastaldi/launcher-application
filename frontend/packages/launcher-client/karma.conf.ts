module.exports = (config) => {
  config.set({
    frameworks: ['jasmine', 'karma-typescript'],
    files: [
      { pattern: 'src/**/*.ts' }
    ],
    preprocessors: {
      '**/*.ts': ['karma-typescript']
    },
    karmaTypescriptConfig: {
      tsconfig: './tsconfig.json'
    },
    reporters: ['mocha', 'karma-typescript'],
    browsers: ['ChromeHeadless'],
    singleRun: true
  });
};
