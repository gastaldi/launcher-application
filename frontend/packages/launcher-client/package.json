{
  "name": "launcher-client",
  "version": "1.0.0-alpha.0",
  "homepage": "https://github.com/fabric8-launcher/launcher-client",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/fabric8-launcher/launcher-client.git"
  },
  "bugs": {
    "url": "https://github.com/fabric8-launcher/launcher-client/issues"
  },
  "main": "build/index.js",
  "module": "build/esm/index.js",
  "types": "build/types/index.d.ts",
  "engines": {
    "node": ">= 8.9.0",
    "npm": ">= 5.5.0"
  },
  "scripts": {
    "build": "rm -rf ./build && rollup -c",
    "test": "jest",
    "semantic-release": "semantic-release"
  },
  "dependencies": {
    "axios": "0.19.0",
    "injection-js": "2.2.1",
    "lscache": "1.3.0",
    "reflect-metadata": "0.1.13"
  },
  "devDependencies": {
    "@types/axios": "0.14.0",
    "@types/lscache": "1.0.29",
    "@types/node": "11.13.4",
    "axios-mock-adapter": "1.16.0",
    "rollup-plugin-json": "4.0.0",
    "rollup-plugin-strip-code": "0.2.6"
  },
  "eslintConfig": {
    "extends": "react-app"
  }
}
