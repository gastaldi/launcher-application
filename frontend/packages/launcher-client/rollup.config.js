import peerDepsExternal from 'rollup-plugin-peer-deps-external';
import filesize from 'rollup-plugin-filesize';
import tslint from 'rollup-plugin-tslint';
import typescript from 'rollup-plugin-typescript2';
import json from 'rollup-plugin-json';

import pkg from './package.json';

const config = {
    input: 'src/index.ts',
    output: [
        {
            file: pkg.main,
            format: 'cjs',
        },
        {
            file: pkg.module,
            format: 'es',
        },
    ],
    external: [
        ...Object.keys(pkg.dependencies || {}),
        ...Object.keys(pkg.peerDependencies || {}),
    ],
    plugins: [
        peerDepsExternal(),
        tslint({
            include: './src/**/*.tsx?'
        }),
        typescript({
            clean: true,
            typescript: require('typescript'),
            verbosity: 0,
            useTsconfigDeclarationDir: true
        }),
        json(),
        // FIXME find another solution to remove the mock (that make it possible to choose from the consumer side)
        /**
        isProduction && stripCode({
            start_comment: 'test-code',
            end_comment: 'end-test-code'
        }),**/
        filesize(),
    ],
};

export default config;
