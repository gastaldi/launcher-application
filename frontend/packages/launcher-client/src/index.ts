/*
 * Public API
 */

export * from './client/launcher.client.factory';
export * from './client/launcher.client';
export * from './client/types';
export * from './client/helpers/launchers';
export * from './client/helpers/mock-helpers';
export * from './client/helpers/preconditions';
