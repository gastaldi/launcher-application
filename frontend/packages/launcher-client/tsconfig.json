{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "src",
    "declaration": true,
    "declarationDir": "build/types"
  },
  "include": [
    "src"
  ]
}
