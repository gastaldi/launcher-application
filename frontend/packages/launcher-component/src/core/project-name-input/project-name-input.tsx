import React, { useState, FormEvent } from 'react';
import { useLauncherClient } from '../../contexts/launcher-client-context';
import { InlineTextInput } from '../inline-text-input/inline-text-input';
import { NAME_REGEX } from '../../flows/launch-flow';

interface ProjectNameInputProps {
  prefix: string;
  value?: string;
  onChange?(value: string, event: FormEvent<HTMLInputElement>): void;
}

export function ProjectNameInput(props: ProjectNameInputProps) {
  const client = useLauncherClient();
  const [warning, setWarning] = useState();

  const isValid = (value: string) => NAME_REGEX.test(value);
  const validateProjectName = async (projectName: string) => {
    if (isValid(projectName)) {
      const result = await client.ocExistsProject(projectName);
      if (result.exists) {
        setWarning('Warning this project exists! Make sure you have write access.');
      } else {
        setWarning(undefined);
      }
    }
  }
  return (
    <InlineTextInput
      type="text"
      id="appname"
      name="appname"
      placeholder="Name of the project"
      aria-label="Application Project name"
      title="Application name"
      value={props.value || ''}
      isValid={isValid(props.value || '')}
      onBlur={() => validateProjectName(props.value || '')}
      warning={warning}
      {...props}
    />
  );
}
