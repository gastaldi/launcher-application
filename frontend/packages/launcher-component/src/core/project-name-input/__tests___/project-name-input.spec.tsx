import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { mockLauncherClient } from 'launcher-client';
import { LauncherDepsProvider } from '../../..';
import { flushPromises } from '../../__tests__/test-helpers';

import { ProjectNameInput } from '../project-name-input';

describe('<ProjectNameInput />', () => {
  const mockClient = mockLauncherClient();
  afterEach(() => {
    cleanup();
  });

  jest.useFakeTimers();

  it('renders the ProjectNameInput correctly', () => {
    const comp = render(
      <LauncherDepsProvider client={mockClient}>
        <ProjectNameInput prefix="New Application:" />
      </LauncherDepsProvider>
    );
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('renders the warning if the project name already exists', async () => {
    const comp = render(
      <LauncherDepsProvider client={mockClient}>
        <ProjectNameInput prefix="New Application:" value="my-project" />
      </LauncherDepsProvider>
    );
    const input = comp.getByLabelText('Application Project name');
    fireEvent.blur(input);
    await flushPromises();

    expect(comp.getByText('Warning this project exists! Make sure you have write access.')).toBeDefined();
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
