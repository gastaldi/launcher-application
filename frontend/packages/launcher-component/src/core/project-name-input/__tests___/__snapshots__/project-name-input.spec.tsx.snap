// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectNameInput /> renders the ProjectNameInput correctly 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <h1
      class=""
      data-pf-content="true"
    >
      New Application:
    </h1>
    <input
      aria-invalid="true"
      aria-label="Application Project name"
      class="pf-c-form-control"
      id="appname"
      name="appname"
      placeholder="Name of the project"
      prefix="New Application:"
      title="Application name"
      type="text"
      value=""
    />
    <span
      style="display: block; color: rgb(240, 171, 0); padding-top: 5px;"
    />
  </div>
</DocumentFragment>
`;

exports[`<ProjectNameInput /> renders the warning if the project name already exists 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <h1
      class=""
      data-pf-content="true"
    >
      New Application:
    </h1>
    <input
      aria-invalid="false"
      aria-label="Application Project name"
      class="pf-c-form-control"
      id="appname"
      name="appname"
      placeholder="Name of the project"
      prefix="New Application:"
      title="Application name"
      type="text"
      value="my-project"
      warning="Warning this project exists! Make sure you have write access."
    />
    <span
      style="display: block; color: rgb(240, 171, 0); padding-top: 5px;"
    >
      Warning this project exists! Make sure you have write access.
    </span>
  </div>
</DocumentFragment>
`;
